Update ACCESS-OM2 run instructions

aidanheerdegen opened the pull request:
## Description

Updates ACCESS-OM2 model run instructions for ACCESS-OM2 Flagship Release.

Fixes #650 

This also works if you include the same text in a commit message.

## Checklist:

- [x] My changes do not break navigation and do not generate new warnings
- [x] I have checked that the links are valid and point to the intended content
- [x] I have checked my code/text and corrected any misspellings
- [x] I have chosen the correct tag for the level of [support provided](https://access-hive.org.au/#support)

aidanheerdegen commented:
This is a draft because I haven't finished it.

aidanheerdegen commented:
Hrm. The `!!! note`  things seemed to not be working. I have got some to work sometimes. 

They're admonitions in mkdocs-material theme

https://squidfunk.github.io/mkdocs-material/reference/admonitions/

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -342,95 +178,361 @@ This will submit a single job to the queue with a total run length of <code>rest
+- _R_ &rarr; Job running
+- _E_ &rarr; Job ending
+
+If there are no jobs listed with your `jobname` (or if no job is listed), your run either successfully completed or was terminated due to an error.
```
Maybe move section ### PBS output files to just here ?

aidanheerdegen replied:
> review the pbs log

Not sure this is the right place for that. I think a dedicated article about PBS and debugging might be a better option and link to it. 

It's not just that I'm trying to get out of work, but I'm concerned about scope-creep for this, and maintainability.

aidanheerdegen replied:
I have moved the PBS files section up though.

anton-seaice replied:
I guess it seemed like there isnt a way to positively confirm the model ran ... Are we saying the creation of an `archive` directory is the sign of success? Otherwise you need to look in the log. (Reviewing it isnt hard ... you open and check it doesn't say Error?)

aidanheerdegen replied:
Oh right. If the model log files `access-om2.err` and `access-om2.out` and the `work` directory are still there then the model hasn't run correctly.

It says this, but maybe not explicitly enough. I'll try and improve this.

aidanheerdegen replied:
Have added a section "Did the model run correctly".

Now realise I have used configuration and model fairly much interchangeably. We need a style guide!

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -342,95 +178,361 @@ This will submit a single job to the queue with a total run length of <code>rest
-<br>
-Outputs and restarts are separated in the respective folders for each model component.
-<br>
+At the end of a successful model run, output files (and restart files) are moved (archived) from the `work` directory to the `archive` directory which is a symbolic link to a directory in the _laboratory_ (`/scratch/$PROJECT/$USER/access-om2/archive`). The model log files are also moved to the `archive` directory. 
```
At the end of a successful model run, output files, restart files and log files are moved from the `work` directory to the `archive` directory.  A symbolic link in the _control_ directory to a directory in the _laboratory_ (`/scratch/$PROJECT/$USER/access-om2/archive`) is provided for convenience.

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -342,95 +178,361 @@ This will submit a single job to the queue with a total run length of <code>rest
+### PBS output files
 
----
+When the model completes PBS writes the standard outout and error streams to two files into the _control_ directory: `jobname.o<job-ID>` and `jobname.e<job-ID>` respectively. This is terminal output that isn't otherwise redirected into model log files.
```
You can archive these files using `payu sweep` which moves them to the `archive` directory.

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -342,95 +178,361 @@ This will submit a single job to the queue with a total run length of <code>rest
+
+### Edit a single {{ model }} component configuration
+
+Each of the [model components] contains configuration options specific to that model that are read in when the model component is running. These options are typically useful to modify the physics used in the model or the input data.
```
the physics used in the model, the input data or the variables saved in the history output.

aidanheerdegen replied:
I've used "diagnostic output". I think "history files" is quite specific to some models, and isn't widely used in other areas. Happy to be told otherwise if I'm wrong.

anton-seaice replied:
DId you mean to make a change here?

to me: diagnostic output means output (often in a log file) used to diagnose if the model is faulty, history output is the variable output you might use for climate analysis

aidanheerdegen replied:
> DId you mean to make a change here?

Yes. I did, but hadn't pushed, thanks.

> to me: diagnostic output means output (often in a log file) used to diagnose if the model is faulty, history output is the variable output you might use for climate analysis

The ocean community use the term "diagnostic output" for output data.

I've gone with

"These options are typically useful to modify the physics used in the model, the input data or the model variables saved in the output files."

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
````diff
@@ -342,95 +178,361 @@ This will submit a single job to the queue with a total run length of <code>rest
+```
+
+Sets environment variables within the PBS job.
+
````
I kind of think the four items above are quite niche (stack size, mpirun arguments, qsub argumnts, environment variables), and unlikely to be relevant in an intro the running the model guide?

aidanheerdegen replied:
I agree, which is why I moved them down into a Miscellaneous section. 

I'll aggregate them, and hide them by default.

anton-seaice reviewed (commented):
Sorry - lots of comments but mostly minor I hope :)

anton-seaice commented:
I assume these commits will get squashed during the merge :)

aidanheerdegen commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -12,291 +12,127 @@ If you are unsure whether {{ model }} is the right choice for your experiment, t
+<terminal-line data="input">payu --version</terminal-line>
+<terminal-line lineDelay="1000">1.1.1</terminal-line>
+</terminal-window>
+
```
Sorry, yes this is a known issue I should have flagged. The deployment was auto-updating the version string. It does now, but was only finally fixed after the last deployment. I have on my TO-DO list to push a new `payu` tag and update `payu` which will mean it will report a proper version. I am am waiting for a few `payu` fixes before doing this.

aidanheerdegen commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -12,291 +12,127 @@ If you are unsure whether {{ model }} is the right choice for your experiment, t
-This is a 1° horizontal resolution configuration with interannual forcing from 1 Jan 1958 to 31 Dec 2018.
-<br>
-To get it on <i>Gadi</i>, create a directory to store the model configuration.Navigate to this directory and clone the GitHub repo in it by running:
+All {{ model }} configurations are available from the [ACCESS-OM2 configs] GitHub repository.  ACCESS-NRI has adapted model configurations from those originally developed by [COSIMA]. 
```
Good point. I'll add a "released" specifier in there. Might have to add a sentence explaining what a released configuration actually *is*. 

There aren't all those configurations yet, but there will be when this is merged.

aidanheerdegen commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -12,291 +12,127 @@ If you are unsure whether {{ model }} is the right choice for your experiment, t
+
+Each configuration is stored as a separate specially named branch in the [ACCESS-OM2 configs] GitHub repository. They are organised like this for administrative convenience: they are easier to manage and keep grouped for testing. Anyone using a configuration is advised to just clone a single branch and not attempt to keep this structure. The [ACCESS-OM2 configs] repo has more information about th ...
+
+The first step is to decide which configuration is required from the twelve available. For example, if the 1° horizontal resolution configuration with repeat-year JRA55 forcing (without bgc) is the required configuration then the [`release-1deg_jra55_ryf`](https://github.com/ACCESS-NRI/access-om2-configs/tree/release-1deg_jra55_ryf) branch is the correct configuration.
```
It's a good point. Thanks.

aidanheerdegen commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -12,291 +12,127 @@ If you are unsure whether {{ model }} is the right choice for your experiment, t
-<pre><code>git clone https://github.com/COSIMA/1deg_jra55_iaf.git</code></pre>
 <terminal-window>
     <terminal-line data="input">mkdir -p ~/access-om</terminal-line>
     <terminal-line data="input">cd ~/access-om</terminal-line>
```
Yep. I can see the value in the terminal thingo, but it doesn't allow for easy copy n' paste of a command, which the code blocks do.

I'll rearrange, but the clone after the terminal and just highlight the command that was run to allow people to copy it out of a code block. See if that works better.

anton-seaice replied:
Maybe ... i think it's just "different"

anton-seaice replied:
Should we just add a comment that thats the purpose of having both - black shows demos, grey is for copy  /paste

aidanheerdegen replied:
Good idea. Have done so.

aidanheerdegen commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -12,291 +12,127 @@ If you are unsure whether {{ model }} is the right choice for your experiment, t
+Furthermore, this separation allows multiple self-resubmitting experiments that share common executables and input data to be run simultaneously.
 
-To change the internal run length, edit the <code>restart_period</code> field in the <code>&date_manager_nml</code> section of the <code>~/access-om/1deg_jra55_iaf/accessom2.nml</code> file:
+To setup the _laboratory_ directory, from the _control_ directory run the:
```
Thanks for all the feedback. To make this easier and more precise [code suggestions](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request) are a great feature.

anton-seaice replied:
Ah yes! I kept selecting the word and clicking reply and being surprised when i got an empy comment !

aidanheerdegen commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -12,291 +12,127 @@ If you are unsure whether {{ model }} is the right choice for your experiment, t
-<br>
-{{ model }} suites run on <a href="https://opus.nci.org.au/display/Help/0.+Welcome+to+Gadi#id-0.WelcometoGadi-Overview" target="_blank"><i>Gadi</i></a> through a <a href="https://opus.nci.org.au/display/Help/4.+PBS+Jobs" target="_blank">PBS job</a> submission managed by <i>payu</i>.
+<terminal-window>
+<terminal-line data="input">cd ~/access-om/1deg_jra55_yaf</terminal-line>
```
Agreed.

aidanheerdegen commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -12,291 +12,127 @@ If you are unsure whether {{ model }} is the right choice for your experiment, t
+As a first step, from within the _control_ directory, it is good practice to run:
 
-<pre><code>payu setup</code></pre>
+    payu setup
```
It *is* really useful, but yeah point taken. I'll move it to the end and call it trouble-shooting or debugging.

aidanheerdegen commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -314,22 +150,22 @@ This will prepare the model run, based on the experiment configuration.
+
+    payu run -f
+
+This will submit a single job to the queue with a run length of `restart_period`.  `restart_period` is defined in the `accessom2.nml` file in the _control_ directory.
```
It is fairly side-effect free. I personally think `payu run` shouldn't care if `work` exists, but I lost that argument. If people use `payu run -f` as a matter of course it will avoid a lot of confused people.

Happy to concede if enough people disagree with me ... ;)

anton-seaice replied:
Ok i haven't thought about the specific useage lot . Just using `-f` without thought gives me alarm bells!

aidanheerdegen commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -342,95 +178,361 @@ This will submit a single job to the queue with a total run length of <code>rest
+If you want to run an {{ model }} configuration multiple times automatically use the option `-n`:
 
-<pre><code>payu run -f -n &lt;number-of-runs&gt;</code></pre>
+    payu run -f -n <number-of-runs>
```
See above. My opinion has changed since then ... :)

aidanheerdegen reviewed (commented):
Ugh. Did this through VS code and it thinks this is a review. Will submit so the comments show up and keep going through the web interface.

aidanheerdegen commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -342,95 +178,361 @@ This will submit a single job to the queue with a total run length of <code>rest
+This will run `number-of-runs` times with a total length of `restart_period * number-of-runs`.
 
-## <pre><code>payu run -f -n 10</code></pre>
+For example, to run a configuration for a total of 50 years with `restart_period = 5, 0, 0` (5 years), the `number-of-runs` should be set to `10`:
```
Yep. Will slightly re-word and add a note linking to that section.

aidanheerdegen commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -342,95 +178,361 @@ This will submit a single job to the queue with a total run length of <code>rest
-## <pre><code>payu run -f -n 10</code></pre>
+For example, to run a configuration for a total of 50 years with `restart_period = 5, 0, 0` (5 years), the `number-of-runs` should be set to `10`:
+
+    payu run -f -n 10
```
I think we've established by now that this my jam.

aidanheerdegen commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -342,95 +178,361 @@ This will submit a single job to the queue with a total run length of <code>rest
-Currently, there is no specific tool to monitor {{ model }} runs.
-<br>
-You can execute the following command to show the status of all your submitted <a href="https://opus.nci.org.au/display/Help/4.+PBS+Jobs" target="_blank">PBS jobs</a>:
+Currently, there is no specific tool to monitor {{ model }} runs. However `payu run` reports the PBS `job-ID`, e.g. `110020843.gadi-pbs`, as the last line to the terminal. `qstat` can be used to query the status of the job, e.g.
```
Agreed.

aidanheerdegen commented:
> I assume these commits will get squashed during the merge :)

Yes, they're not at all functional as commits.

aidanheerdegen commented:
> Sorry - lots of comments but mostly minor I hope :)

No, thank you. A fresh perspective is much appreciated.

If you have time and while it is fresh could you take a look at the amended version to make sure you're happy with the changes and if you think of anything else that'd be great.

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -12,291 +12,228 @@ If you are unsure whether {{ model }} is the right choice for your experiment, t
-    <terminal-line>remote: Total 14715 (delta 3383), reused 3379 (delta 3377), pack-reused 11314</terminal-line>
-    <terminal-line>Receiving objects: 100% (14715/14715), 35.68 MiB | 18.11 MiB/s, done.</terminal-line>
-    <terminal-line>Resolving deltas: 100% (10707/10707), done.</terminal-line>
+    <terminal-line data="input" directory="~/access-om">payu clone -b expt -B release-1deg_jra55_ryf https://github.com/COSIMA/1deg_jra55_ryf.git 1deg_jra55_ryf</terminal-line>
```
```suggestion
    <terminal-line data="input" directory="~/access-om">payu clone -b expt -B release-1deg_jra55_ryf https://github.com/ACCESS-NRI/access-om2-configs.git 1deg_jra55_ryf</terminal-line>
```

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -12,291 +12,228 @@ If you are unsure whether {{ model }} is the right choice for your experiment, t
+
+In the example above the `payu clone` command clones the 1° repeat-year JRA55 configuration (` -B release-1deg_jra55_ryf`) 
+to a new experiment branch (`-b expt`) to a directory named `1deg_jra55_ryf`: 
+
```
```suggestion
     mkdir -p ~/access-om
     cd ~/access-om2
```

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -12,291 +12,127 @@ If you are unsure whether {{ model }} is the right choice for your experiment, t
-<div class="note">
-    The internal run length (controlled by <code>restart_period</code>) can be different from the total run length. Also, while <code>restart_period</code> can be reduced, it should not be increased to more than 5 years to avoid errors. For more information about the difference between internal run and total run lengths, or how to run the model for more than 5 years, refer to the section <a href= ...
-</div>
+    payu init
```
Ping?

aidanheerdegen replied:
It actually isn't currently, though that has brought this up as a potential area that needs fixing.

https://github.com/payu-org/payu/issues/428

So it should stay until the issue is fixed.

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -12,291 +12,228 @@ If you are unsure whether {{ model }} is the right choice for your experiment, t
+
+This will prepare the model run: create the ephemeral `work` directory based on the experiment configuration, generate manifests and report some useful information to the user, such as the location of the _laboratory_ where the `work` and `archive` directories are located.
 <terminal-window>
 <terminal-line data="input">payu setup</terminal-line>
```
```suggestion
<terminal-line data="input">payu setup -f</terminal-line>
```

aidanheerdegen replied:
Thanks! Glad you've to see you've joined team force.

anton-seaice replied:
I haven't. But i like consistency !

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -314,123 +251,256 @@ This will prepare the model run, based on the experiment configuration.
+# References
+
+- [COSIMA]
+- [Kiss et al. (2020)](http://doi.org/10.5194/gmd-13-401-2020")
```
```suggestion
- [Kiss et al. (2020)](http://doi.org/10.5194/gmd-13-401-2020)
```

anton-seaice replied:
This one :). Do this one :)

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -314,123 +251,256 @@ This will prepare the model run, based on the experiment configuration.
+
+- [COSIMA]
+- [Kiss et al. (2020)](http://doi.org/10.5194/gmd-13-401-2020")
+- https://github.com/COSIMA/access-om2/
```
```suggestion
- https://github.com/access-nri/access-om2/
```

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -314,123 +251,256 @@ This will prepare the model run, based on the experiment configuration.
+
+Each of the [model components] contains configuration options specific to that model that are read in when the model component is running. These options are typically useful to modify the physics used in the model, the input data or the variables saved in the diagnostic output. 
+These configuration options are are specified in files in a subfolder of the _control_ directory, named the same as the submodel's name in the `config.yaml` `submodel` section (e.g., configuration options for the _ocean_ submodel are in the `~/access-om/1deg_jra55_iaf/ocean` directory).
+To modify these options please refer to the User Guide of each individual model component.
```
Do we have a neat way to link to these? I think not?

aidanheerdegen replied:
Yeah that's a bit of a cop-out eh? Off the top of my head we don't provide any such thing. We should. Don't know if we can in any reasonable time-frame, but should make it a TO-DO.

anton-seaice replied:
Yeah - i think its hard though. Its not obvious where cice 5 documentation is. e.g. its not in the cice readthedocs history, and then it was modified a fair bit by COSIMA. (as an example)

aidanheerdegen replied:
It is a lot tougher for legacy models we're taking over for sure. But it reasonable that we make some effort to collect together what is out there.

aidanheerdegen replied:
I have made issues in the model component repros to try and get this happening.

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -10,6 +10,10 @@ Before running {{ model }}, you need to fulfil general prerequisites outlined in
 
 If you are unsure whether {{ model }} is the right choice for your experiment, take a look at the overview of [ACCESS Models](/models).
 
+### Conventions
```
Is this better in a note block ?

aidanheerdegen replied:
Good suggestion. Thanks. Done.

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -10,293 +10,246 @@ Before running {{ model }}, you need to fulfil general prerequisites outlined in
+
+In the example above the `payu clone` command clones the 1° repeat-year JRA55 configuration (` -B release-1deg_jra55_ryf`) 
+to a new experiment branch (`-b expt`) to a directory named `1deg_jra55_ryf`: 
+     mkdir -p ~/access-om
```
```suggestion
    mkdir -p ~/access-om
```

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -10,293 +10,246 @@ Before running {{ model }}, you need to fulfil general prerequisites outlined in
+In the example above the `payu clone` command clones the 1° repeat-year JRA55 configuration (` -B release-1deg_jra55_ryf`) 
+to a new experiment branch (`-b expt`) to a directory named `1deg_jra55_ryf`: 
+     mkdir -p ~/access-om
+     cd ~/access-om2
```
```suggestion
    cd ~/access-om2
```

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -10,293 +10,246 @@ Before running {{ model }}, you need to fulfil general prerequisites outlined in
-After editing the configuration, you are ready to run {{ model }}.
-<br>
-{{ model }} suites run on <a href="https://opus.nci.org.au/display/Help/0.+Welcome+to+Gadi#id-0.WelcometoGadi-Overview" target="_blank"><i>Gadi</i></a> through a <a href="https://opus.nci.org.au/display/Help/4.+PBS+Jobs" target="_blank">PBS job</a> submission managed by <i>payu</i>.
+    work
```
```suggestion
    work/
```

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -10,293 +10,246 @@ Before running {{ model }}, you need to fulfil general prerequisites outlined in
+Model components are separated into subdirectories within the output and restart directories.
+
+<terminal-window>
+<terminal-line data="input">cd ~/access-om/1deg_jra55_yaf<</terminal-line>
```
```suggestion
<terminal-line data="input">cd ~/access-om/1deg_jra55_ryf<</terminal-line>
```

anton-seaice commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -314,123 +267,256 @@ This will prepare the model run, based on the experiment configuration.
+
+- [COSIMA]
+- [Kiss et al. (2020)](http://doi.org/10.5194/gmd-13-401-2020)
+- https://github.com/access-nri/access-om2/
```
I am not sure why https://github.com/access-nri/access-om2/ is not getting formatted as a link in the build preview

aidanheerdegen replied:
Fixed.

anton-seaice reviewed (approved):
I left a few more very minor formatting things, but happy to approve at this point :)

anton-seaice commented:
Just this bit of formatting is off:

![Screen Shot 2024-03-19 at 1 11 55 pm](https://github.com/ACCESS-NRI/access-hive.org.au/assets/79179784/010659e7-be42-448e-8975-4ea7518e035f)

aidanheerdegen commented:
> Just this bit of formatting is off

Fixed. Thanks. I believe I've also addressed all the out-standing comments.

And thank you for the thorough and time-consuming review.

anton-seaice commented:
No worries . All looks good Aidan.

minghangli-uni commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -342,95 +178,361 @@ This will submit a single job to the queue with a total run length of <code>rest
+While the model is running, _payu_ saves the model standard output and standard error in the `access-om2.out` and `access-om2.err` log files in the _control_ directory. You can examine the contents of these files to check on the status of a run as it progresses.
 
-### Model Live Diagnostics
+At the end of a successful run these log files are archived. If they remain in the _control_ directory after the PBS job for a run has completed this is an indication the run has failed.
```
For first time users, I suggest to be more explicit. We are spooning this to users in a straightforward way. :)

```suggestion
At the end of a successful run these log files are archived to the `archive` directory and will not be found in the _control_ directory. If they remain in the _control_ directory after the PBS job for a run has completed this is an indication the run has failed.
```

minghangli-uni commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -342,95 +178,361 @@ This will submit a single job to the queue with a total run length of <code>rest
+
+### Change run length
+
+One of the most common changes is to how long the model runs. For example it is common when debugging changes to a model to want to reduce the run length to reduce the amount of resources wasted and get faster feedback on changes.
```
```suggestion
One of the most common changes is to adjust the duration of the model run. For example when debugging changes to a model, it is common to reduce the run length to minimise resource consumption and return faster feedback on changes.
```

minghangli-uni commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -342,95 +178,361 @@ This will submit a single job to the queue with a total run length of <code>rest
+
+One of the most common changes is to how long the model runs. For example it is common when debugging changes to a model to want to reduce the run length to reduce the amount of resources wasted and get faster feedback on changes.
+
+To change the run length, edit the `restart_period` field in the `&date_manager_nml` section of the `~/access-om/1deg_jra55_iaf/accessom2.nml` file:
```
iaf->ryf

minghangli-uni commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -342,95 +178,361 @@ This will submit a single job to the queue with a total run length of <code>rest
+
+This section specifies the submodels and configuration options required to execute the model correctly.
+
+Each submodel contains additional configuration options that are read in when the submodel is running. These options are specified in the subfolder of the _control_ directory, whose name matches the submodel's _name_ (e.g., configuration options for the `ocean` submodel are in the `~/access-om/1deg_jra55_iaf/ocean` directory).
```
iaf->ryf

minghangli-uni commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -342,95 +178,361 @@ This will submit a single job to the queue with a total run length of <code>rest
+### Edit a single {{ model }} component configuration
+
+Each of the [model components] contains configuration options specific to that model that are read in when the model component is running. These options are typically useful to modify the physics used in the model or the input data.
+These configuration options are are specified in files in a subfolder of the _control_ directory, named the same as the submodel's name in the `config.yaml` `submodel` section (e.g., configuration options for the _ocean_ submodel are in the `~/access-om/1deg_jra55_iaf/ocean` directory).
```
iaf->ryf

minghangli-uni commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -314,22 +150,22 @@ This will prepare the model run, based on the experiment configuration.
-To run {{ model }} configuration for one internal run length (controlled by <code>restart_period</code> in the <code>~/access-om/1deg_jra55_iaf/accessom2.nml</code> file), execute:
+To run an {{ model }} configuration: 
+
+    payu run -f
```
I suggest avoiding the use of `-f` in the guide. But it could be mentioned as an option, similar to what was done below.

If a previous run fails and leaves data in work, running payu run would trigger a 'pause', indicating that there was an issue with the previous run. This would guide the user to address any errors encountered in the former run. Otherwise, careless users might assume that all their runs were successful, only to find no data in archive when attempting to post-process.

aidanheerdegen replied:
Ok. I bow to the collective wisdom.  Will remove them all.

minghangli-uni reviewed (commented):
@anton-seaice has conducted a thorough review of the OM2 instructions. I merely stand on the shoulders of @aidanheerdegen and @anton-seaice to propose some very minor changes. All looks good to me!

aidanheerdegen commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -314,22 +150,22 @@ This will prepare the model run, based on the experiment configuration.
 ### Run configuration
 
-To run {{ model }} configuration for one internal run length (controlled by <code>restart_period</code> in the <code>~/access-om/1deg_jra55_iaf/accessom2.nml</code> file), execute:
+To run an {{ model }} configuration: 
```
The jinga template will replace `{{ model }}` with ACCESS-OM2, so the sentence becomes:

> To run an ACCESS-OM2 configuration

To be honest it could be removed completely, and just be

> To run a configuration.

I will make that change.

aidanheerdegen commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -342,95 +178,361 @@ This will submit a single job to the queue with a total run length of <code>rest
+### Run configuration multiple times
 
-If you want to run {{ model }} configuration for multiple internal run lengths (controlled by <code>restart_period</code> in the <code>~/access-om/1deg_jra55_iaf/accessom2.nml</code> file), use the option <code>-n</code>:
+If you want to run an {{ model }} configuration multiple times automatically use the option `-n`:
```
See above. Will remove explicit mention of the model.

aidanheerdegen commented:
Thanks for the review @minghangli-uni. Really useful suggestions and I believe I have incorporated them all.

minghangli-uni reviewed (approved):
I've taken another look at it. All good to me

truth-quark commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -1,302 +1,262 @@
+In the example above the `payu clone` command clones the 1° repeat-year JRA55 configuration (` -B release-1deg_jra55_ryf`) 
+to a new experiment branch (`-b expt`) to a directory named `1deg_jra55_ryf`: 
+
+    mkdir -p ~/access-om
+    cd ~/access-om2
```
We've got a typo/inconsistency between dir names here: `access-om` & `access-om2`

aidanheerdegen replied:
Made it consistent throughout document to be `~/access-om2`

truth-quark commented on docs/models/run-a-model/run-access-om.md:
```diff
@@ -1,302 +1,262 @@
-Each of <a href="/models/configurations/access-om/#model-components">{{ model }} components</a> contains additional configuration options that are read in when the model component is running. These options are typically useful to modify the physics used in the model or the input data.
-They are specified in the subfolder of the <i>control</i> directory, whose name matches the submodel's name as specified in the <code>config.yaml</code> <code>submodel</code> section (e.g., configuration options for the <code>ocean</code> submodel are in the <code>~/access-om/1deg_jra55_iaf/ocean</code> directory).
-To modify these options please refer to the User Guide of each individual model component.
+    payu init
```
From the view of a new user, does the init command _need_ to be run?

aidanheerdegen replied:
Can you confirm that it isn't necessary? I don't believe it is, but would be good to have that tested. Note that if the laboratory already exists to do a proper test the laboratory directories would have to be deleted, or the test run using a different project so there was not an existing laboratory.

aidanheerdegen replied:
I have [confirmed `payu init` is not required](https://github.com/payu-org/payu/issues/428#issuecomment-2050853660). Removed.

aidanheerdegen commented:
@atteggiani I have pinged you for a review. If you can't manage it today can you let me know, as I'd like this to go live this afternoon.

Thanks!

atteggiani commented:
Hi @aidanheerdegen,
I am checking the PR and content-wise is all good, but there are many (little) things that need to be modified for the style and consistency with the other pages.

My plan is the following:
1. I create a branch called `davide/review-PR-OM2` from this branch (`aidanheerdegen/issue650`)
2. I do the modifications in my new branch
3. I open a PR to merge my new branch (`davide/review-PR-OM2`) into this branch (`aidanheerdegen/issue650`)
4. You review the PR so we merge the two branches
5. We merge this (updated) PR to development

The whole process should not take more than 1-2 hours.

atteggiani reviewed (approved):
Approving with updates that will take place next week.
